migrations/001.do.init.sql

```diff
@@ -22,4 +22,23 @@ CREATE TABLE hunt (
     enemy_id INT NOT NULL references enemies(id) ON DELETE CASCADE,
     PRIMARY KEY(character_id, enemy_id)
 );
-ALTER TABLE players ADD COLUMN active_character INT REFERENCES characters(id) ON DELETE SET NULL;
+CREATE TYPE enum_material AS ENUM('Dragon Talon', 'Dragon Hide', 'Dragon Fang');
+CREATE TABLE materials (
+    type enum_material NOT NULL,
+    quantity (int),
+    PRIMARY KEY(player_id, material_id)
+);
+CREATE TYPE enum_geartype AS ENUM('DragonsBite');
+CREATE TABLE equipment (
+    id SERIAL PRIMARY KEY,
+    player_id INT NOT NULL references players(id) ON DELETE CASCADE,
+    type enum_geartype NOT NULL;
+);
+-- CREATE TYPE enum_mod AS ENUM('PhysAtk', 'FireAtk', 'IceAtk', 'ShockAtk', 'PhysDef', 'FireDef', 'IceDef', 'ShockDef');
+CREATE TABLE equipment_mods (
+    mods JSONB,
+    PRIMARY KEY(player_id, equipment_id),
+);
```

Comment:
Missing the columns of the primary key again. Actually in this case, you need a serial primary key - there is no player for an equipment mod, just an equipment, but since there will be multiple mods for each equipment, the mods will need their own ids.

Also, when in doubt just put `NOT NULL` on the column, otherwise we gotta deal with null which is a pain.

```suggestion
CREATE TABLE equipment_mods (
    id SERIAL PRIMARY KEY,
    equipment_id INT NOT NULL REFERENCES equipment (id) ON DELETE CASCADE,
    mods JSONB NOT NULL
);
```

Reply:
I was going to say that it's possible for weapons to not have mods... but let's just scrap that entirely and make it so that when the item is created, it comes with mods right off the bat.

Reply:
Don't forget `null` is a valid JSON value.

Also worth considering: do we need a whole table for mods, with one mod per entry, or do we just make the JSONB hold an array and put it directly on the equipment?